lwk_wollet/src/pset_create.rs

```diff
@@ -245,4 +254,58 @@ mod test {
         let network = ElementsNetwork::Liquid;
         assert!(validate_address(testnet_address, network).is_err())
     }
+
+    #[test]
+    fn test_add_input_exceeds_limit() {
+        let wollet = test_wollet_with_many_transactions();
+
+        let mut pset = PartiallySignedTransaction::default();
+        for _ in 0..SECP256K1_SURJECTIONPROOF_MAX_N_INPUTS {
+            pset.add_input(Default::default());
+        }
+
+        let mut inp_txout_sec = HashMap::new();
+        let mut inp_weight = 0usize;
+        let dummy_utxo = wollet.utxos().unwrap()[0].clone();
+        let result = wollet.add_input(&mut pset, &mut inp_txout_sec, &mut inp_weight, &dummy_utxo);
+
+        match result {
+            Err(Error::TooManyInputs(count)) => {
+                assert_eq!(count, SECP256K1_SURJECTIONPROOF_MAX_N_INPUTS);
+            }
+            _ => panic!("Expected TooManyInputs error"),
+        }
+    }
+
+    #[test]
```

Comment:
I think this could all be in one test. 

Add 256 inputs in a loop (as you are already doing), asserting that each one is added Ok. 

Then after the loop add the 257th input and assert on the Err